### PR TITLE
`ReflectionClass::getConstant()`, `getConstants()` and `getImmediateConstants()` more consistent with methods and properties methods

### PR DESCRIPTION
**Pull request by kukulich**
- Return `ReflectionClassConstant` instances
- Have filter parameter

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -633,74 +633,38 @@ public function hasMethod(string $methodName): bool
...
-     * Get the value of the specified class constant.
-     *
-     * Returns null if not specified.
+     * up parent classes etc.), with keys as constant names and values as {@see ReflectionClassConstant} objects.
      *
-     * @return scalar|array<scalar>|null
+     * @return array<string, ReflectionClassConstant> indexed by name
      */
-    public function getConstant(string $name): string|int|float|bool|array|null
+    public function getImmediateConstants(int|null $filter = null): array
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php:640]: If we only stick to `int $filter`, we should document its bitmask type precisely

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -633,74 +633,38 @@ public function hasMethod(string $methodName): bool
...
-    }
-
-    /**
-     * Get the value of the specified class constant.
-     *
-     * Returns null if not specified.
+     * up parent classes etc.), with keys as constant names and values as {@see ReflectionClassConstant} objects.
      *
-     * @return scalar|array<scalar>|null
+     * @return array<string, ReflectionClassConstant> indexed by name
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php:638]: `non-empty-string` keys?

File: src/Reflection/ReflectionClass.php [hunk 3]
```diff
@@ -633,74 +633,38 @@ public function hasMethod(string $methodName): bool
...
-    }
-
-    /**
-     * Get an associative array of only the constants for this specific class (i.e. do not search
-     * up parent classes etc.), with keys as constant names and values as {@see ReflectionClassConstant} objects.
-     *
-     * @return array<string, ReflectionClassConstant> indexed by name
-     */
-    public function getImmediateReflectionConstants(): array
+    public function getConstant(string $name): ReflectionClassConstant|null
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php:665]: We should only accept `non-empty-string` as name

File: src/Reflection/ReflectionClass.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionClass.php:640]: @Ocramius Can I prepare everything in next PR? I will change it for methods and properties as well.

Comment (Ocramius) [reply src/Reflection/ReflectionClass.php:640]: Sure 👍

**Pull request merged**